File: SyntaxNode.rb
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# SyntaxNode.rb

class SyntaxNode
    attr_reader :type
    attr_writer :type
    
    public
        def initialize( type )
            @type = type
        end
        
        def visitNode()
        end
        
        def to_s()
            str = @type.to_s
            
            return str
        end
end

File: SyntaxNode.py
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# SyntaxNode.py

class SyntaxNode( object ):
    
    def __init__( self, nodeType ):
        self.nodeType = nodeType
    
    def visitNode( self, log, rssInfo ):
        pass
    
    def __str__( self ):
        retStr = str( self.nodeType )
        
        return retStr
